### ISSUE TITLE
Numbered list numbering problem with text writer

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1108]
When my documentation has a numbered list that spans headings, the HTML and latexpdf outputs are as expected. The text writer's output restarts the numbering from 1.

For example, with the following source:

```
Test
====

1. one

Sub-section
-----------

2. two

Sub-Sub-Section
```
1. three

```

The output of the text writer is:

```

Test

---
1. one
# Sub-section
1. two
## Sub-Sub-Section
1. three

```

----------------------------------------
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1108
- Originally reported by: Anonymous
- Originally created at: 2013-02-14T02:52:55.326
```

**Comment by shimizukawa** [comment]
_From Ewan Edwards on 2013-02-14 16:32:15+00:00_

It looks like the solution to this problem is to adjust the visit_enumerated_list() method in the text writer from:

```
    def visit_enumerated_list(self, node):
        self.list_counter.append(0)
```

to something like:

```
    def visit_enumerated_list(self, node):
        if 'start' in node:
            self.list_counter.append(node['start'] - 1)
        else:
            self.list_counter.append(0)
```

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2013-02-20 20:41:21+00:00_

Closes #1108: [text] Correctly number enumerated lists with non-default start values (based on patch by Ewan Edwards).

→ <<cset 66d799bb003412555550690f0d7e36719537c388>>

**Comment by shimizukawa** [comment]
_From Jon Waltman on 2013-02-20 20:42:00+00:00_

Thanks
